xline/src/storage/auth_store/backend.rs

```diff
@@ -726,140 +755,124 @@ impl AuthStoreBackend {
                 debug!("Sync AuthUserChangePasswordRequest {:?}", req);
-                self.sync_user_change_password_request(req);
+                self.sync_user_change_password_request(req)?;
             }
             RequestWrapper::AuthUserDeleteRequest(req) => {
                 debug!("Sync AuthUserDeleteRequest {:?}", req);
-                self.sync_user_delete_request(&req);
+                self.sync_user_delete_request(&req)?;
             }
             RequestWrapper::AuthRoleAddRequest(req) => {
                 debug!("Sync AuthRoleAddRequest {:?}", req);
-                self.sync_role_add_request(req);
+                self.sync_role_add_request(req)?;
             }
             RequestWrapper::AuthRoleGetRequest(req) => {
                 debug!("Sync AuthRoleGetRequest {:?}", req);
-                Self::sync_role_get_request(&req);
             }
             RequestWrapper::AuthRoleGrantPermissionRequest(req) => {
                 debug!("Sync AuthRoleGrantPermissionRequest {:?}", req);
-                self.sync_role_grant_permission_request(req);
+                self.sync_role_grant_permission_request(req)?;
             }
             RequestWrapper::AuthRoleRevokePermissionRequest(req) => {
                 debug!("Sync AuthRoleRevokePermissionRequest {:?}", req);
-                self.sync_role_revoke_permission_request(&req);
+                self.sync_role_revoke_permission_request(&req)?;
             }
             RequestWrapper::AuthRoleListRequest(req) => {
                 debug!("Sync AuthRoleListRequest {:?}", req);
-                Self::sync_role_list_request(&req);
             }
             RequestWrapper::AuthRoleDeleteRequest(req) => {
                 debug!("Sync AuthRoleDeleteRequest {:?}", req);
-                self.sync_role_delete_request(&req);
+                self.sync_role_delete_request(&req)?;
             }
             RequestWrapper::AuthenticateRequest(req) => {
                 debug!("Sync AuthenticateRequest {:?}", req);
-                Self::sync_authenticate_request(&req);
             }
             _ => {
                 unreachable!("Other request should not be sent to this store");
             }
         }
-        self.header_gen.revision()
+        Ok(self.header_gen.revision())
     }
 
     /// Sync `AuthEnableRequest` and return whether authstore is changed.
-    fn sync_auth_enable_request(&self, _req: &AuthEnableRequest) {
+    fn sync_auth_enable_request(&self, _req: &AuthEnableRequest) -> Result<(), ExecuteError> {
         if self.is_enabled() {
-            return;
+            return Ok(());
         }
         let revision = self.revision.next();
-        self.put(AUTH_ENABLE_KEY.to_vec(), vec![1], revision, 0);
+        self.put(AUTH_ENABLE_KEY.to_vec(), vec![1], revision, 0)?;
         self.enabled.store(true, AtomicOrdering::Release);
-        self.create_permission_cache();
+        self.create_permission_cache()?;
```

Comment:
`create_permission_cache` returns `Result<(), ExecuteError>`, so we can use `self.create_permission_cache()` instead of `self.create_permission_cache()?; Ok(())`

Reply:
fixed